core/src/main/java/org/apache/calcite/rel/rules/FilterProjectTransposeRule.java

```diff
@@ -118,6 +121,15 @@ public void onMatch(RelOptRuleCall call) {
     final RelBuilder relBuilder = call.builder();
     RelNode newFilterRel;
     if (copyFilter) {
+      // unknownAsFalse => true since in the WHERE clause:
+      // 1>null evaluates to unknown and WHERE unknown behaves exactly like WHERE false
+      RexSimplify simplifier =
+          new RexSimplify(relBuilder.getRexBuilder(), RelOptPredicateList.EMPTY, true,
+              Util.first(call.getPlanner().getExecutor(), RexUtil.EXECUTOR));
+      newCondition = simplifier.simplify(newCondition);
+      // In case the condition is always false we could remove entirely the filter but since the API
+      // indicates that the filter operator must be copied this simplification is not performed
```

Comment:
@zabetak , I don't think the API says the rule MUST copy filter.
I think the idea there is "if the rule happens to create a filter, it should use copy to result in exactly the same kind of filter class".

In other words, it should be perfectly fine to ignore the filter altogether.